code/campaign/src/main/java/com/adobe/marketing/mobile/campaign/FullScreenMessage.java

```diff
@@ -265,13 +263,26 @@ private Map<String, String> createCachedResourcesMap() {
                 currentAssetNumber++;
             }
 
-            // if assetValue is still null, none of our urls have been cached, so we check for a bundled asset
+            // if assetValue is still null then the specified remote url has not been cached.
+            // if a bundled asset is available, use it as a fallback by caching it for use later when the message is displayed.
             if (StringUtils.isNullOrEmpty(assetValue)) {
                 assetValue = currentAssetArray.get(currentAssetArrayCount - 1);
                 boolean isLocalImage = !UrlUtils.isValidUrl(assetValue);
 
                 if (isLocalImage) {
-                    fallbackImagesMap.put(assetUrl, assetValue);
+                    try {
+                        final String cacheName = CampaignConstants.CACHE_BASE_DIR + File.separator + CampaignConstants.MESSAGE_CACHE_DIR;
+                        final String assetCacheUrl = CampaignConstants.LOCAL_ASSET_URI + assetUrl;
+                        final InputStream bundledFile = ServiceProvider.getInstance().getAppContextService().getApplicationContext().getAssets().open(assetValue);
+                        cacheService.set(cacheName, assetCacheUrl, new CacheEntry(bundledFile, CacheExpiry.never(), null));
+                        fallbackImagesMap.put(assetCacheUrl, cacheName);
+                        bundledFile.close();
+                    } catch (final IOException exception) {
+                        Log.debug(CampaignConstants.LOG_TAG, SELF_TAG,
+                                "createCachedResourcesMap - Exception occurred reading bundled asset: %s.", exception.getMessage());
+                        break;
```

Comment:
Shouldn't this be `continue` to continue processing other assets in the array?

Reply:
actually just removed it as its the end of the for loop